docs/configuration/README.md

```diff
@@ -31,3 +31,34 @@ var chartDifferentHoverMode = new Chart(ctx, {
     }
 });
 ```
+
+## Dataset Configuration
+
+Options may be configured directly on the dataset. The default option values can be changed globally across a dataset type. E.g. an option may be given a default value for all `line` datasets as demonstrated below.
+
+Chart.js merges user-specified dataset configuration with the dataset defaults. This way you can be as specific as you would like in your individual dataset configuration, while still changing the defaults for all datasets where applicable. The dataset general options are defined in `Chart.defaults.global.datasets.type` where `type` corresponds to the dataset type. You may also set `chart.options.datasets.type` to set the default at the chart level instead of globally. Dataset options take precedence over element options. If you set a dataset type default option, it will override all corresponding element options. The defaults for each dataset type are discussed in the documentation for that chart type.
```

Comment:
I think `options.datasets[type]` is less confusing than `options.datasets.type`.

Also, I find this paragraph a bit hard to assimilate, maybe we could simplify it this way (or similar):

> The dataset options can be changed at 3 different levels and are evaluated with the following priority:
>
>- per dataset: `dataset.*`
>- per chart: `options.datasets[type].*`
>- or globally: `Chart.defaults.global.datalabels[type].*`
>
>where `type` corresponds to the dataset type.
>
> **Note:** dataset options take precedence over element options.